avm/utilities/pipelines/e2eValidation/regionSelector/Get-AzAvailableResourceLocation.ps1

```diff
@@ -85,13 +96,20 @@ function Get-AzAvailableResourceLocation {
     } |  Select-Object -ExpandProperty Location
     Write-Verbose "Available Locations: $($locations | ConvertTo-Json)"
 
-
-    $index = Get-Random -Maximum ($locations.Count)
-    Write-Verbose "Generated random index [$index]"
-
-    $location = $locations[$index]
-
+    $filteredAllowedLocations = @()
+    $filteredAllowedLocations = @($locations | Where-Object { $_ -in $AllowedRegionsList })
+    Write-Verbose "Filtered allowed locations: $($filteredAllowedLocations | ConvertTo-Json)"
+
+    if ($filteredAllowedLocations.Count -gt 0) {
+      $index = Get-Random -Maximum ($filteredAllowedLocations.Count)
+      Write-Verbose "Generated random index [$index]"
+      $location = $filteredAllowedLocations[$index]
+    }
+    else {
```

Comment:
Hey @ChrisSidebotham, not sure I got the intention of the else block, could you please elaborate?

Reply:
If the list of location is greater than 0, then we generate a random index. if it is 0 then we have found 1 location and therefore pass that directly, otherwise PowerShell errors as the Array contains 1 item

Reply:
Not sure I follow :D 

If `filteredAllowedLocations` is empty, that means `$locations | Where-Object { $_ -in $AllowedRegionsList }` was empty. Wouldn't that mean no location is available and we'd be assigning `$location = @()` ?

Reply:
This is actually left over from before I updated line 184, Removed the if, else block now as not needed

Reply:
Not sure I follow what you mean here? 

These test cases have been ran 3/4 times in the fork with no failures relating to location of deployment. 

If a resource is added and does not have a supported location then we should expand the allowList to include one region in which it is avaialble. Although this should be covered between `eastus` and `eastasia`

Reply:
I got that tests are successful and In agree that's the most important part to stabilize the current situation. That's what I mean for quick win.

As the implementation does not cover for the possible empty $filteredAllowedLocations array, I'm just suggesting a try catch block to handle the error.

Reply:
Don't have anything more to add. Will leave the final call to you @eriqua :)